apps/meteor/app/livechat/server/lib/Contacts.ts

```diff
@@ -248,6 +257,36 @@ export async function updateContact(params: UpdateContactParams): Promise<ILivec
 	return updatedContact;
 }
 
+export async function getContacts(params: GetContactsParams): Promise<PaginatedResult<{ contacts: ILivechatContact[] }>> {
+	const { searchText, count, offset, sort } = params;
+
+	const searchRegex = escapeRegExp(searchText || '');
+	const match: Filter<ILivechatContact & RootFilterOperators<ILivechatContact>> = {
+		$or: [
+			{ name: { $regex: searchRegex }, $options: 'i'},
+			{ emails: { $regex: searchRegex }, $options: 'i'},
+			{ phones: { $regex: searchRegex }, $options: 'i'},
+		]
```

Comment:
the contacts model has no indexes and this would likely require 3. not sure if this would be premature but Users model has a case insensitive index for user email. maybe the same can be done here